### ISSUE TITLE
TileMap Merging Improvements!

### ISSUE THREAD

**Issue by yaseenrehan123** [godotengine/godot-proposals#8804]
### Describe the project you are working on

I am working on a 2d platformer game.

### Describe the problem or limitation you are having in your project

I want to propose better tileset merging options. If you want to add something or a new tile to your already existing tilemap it creates or saves it as a new tileset and than if we delete the previous tileset the progress and all its tiles is gone. In short their should be a option to simply add the new tile to the exisiting tileset without creting a new tileset, and this will ensure that the progress is not resetted cause if you merge the tilesets now they save it as a new tileset and the progress(already exisitng tiles in levels are all cleared)

https://github.com/godotengine/godot-proposals/assets/109503120/2b51889f-f3fe-4a58-bd67-35ff3f14950e

Summary:
A feature to merge the new tile in the existing tileset without creating a new tileset

### Describe the feature / enhancement and how it helps to overcome the problem or limitation

This can help alot if a new tile is needed to be added in the current tileset, Also fixes the previous bug/problem of tilemap clearing all the tiles after merging. Can help if the tileset is not finalized and needs to add/remove tiles from it for experimentation.
Basically it allows users to edit the tilesets more easily

### Describe how your proposal will work, with code, pseudo-code, mock-ups, and/or diagrams

![Screenshot (138)](https://github.com/godotengine/godot-proposals/assets/109503120/3727786f-17a8-459b-b5ac-88efde3ff23e)

In the tileset window thier can be another option or button of "merge in existing tileset" to merge the new tile or png in the existing tileset.

### If this enhancement will not be used often, can it be worked around with a few lines of script?

Potentially yes?, as i said above if the enhancement isnt used we can just remove the new code or the button that triggers it.

### Is there a reason why this should be core and not an add-on in the asset library?

I think its pretty important feature to allow more flexibility for the editor and the engine allowing users to more easily create vast worlds with large tilesets and easily customize their tilesets to add or remove specific tiles.

**Comment by KoBeWi** [comment]
But to add new tiles you can just edit the original image, why do you need to merge something?

**Comment by yaseenrehan123** [comment]
> Member

merging would be more effective as right now i dont think their is  proper way to edit a tileset inside godot thats why,currently if we merge the tileset it would create it as a new tileset so now you will have 2 tilsets and if you delete the original tileset(the previous tileset) than all the tiles placed with tha tileset will be gone.
and you will have to place them over with the new merged tileset which is a pain, so thats why i wanted to propose to merge in a way to keep the original tileset but add new tiles

**Comment by KoBeWi** [comment]
> merging would be more effective as right now i dont think their is proper way to edit a tileset inside godot thats why

But there is no way to *create* a TileSet either. You need to create and edit images in external editor, and there is no reason to create two images and merge them instead of editing one image.

**Comment by yaseenrehan123** [comment]
even if we like edit the image and import it in godot and mke it into  tileset, wont the previous progress be lost, cause after all its still a new image and godot would consider it as a new tileset so if you update your image by editing it, it would make a new tileset? and all the tiles will either be setted wrongly than or will no appear

**Comment by KoBeWi** [comment]
No, you can freely edit the image and change its size and there is no data loss. If you make the image bigger, you can normally create new tiles in the added space. If there image is smaller, your tiles will stay outside the texture until you run cleanup.
